src/storage/shared/identity-storage.ts

```diff
@@ -44,11 +44,20 @@ export class IdentityStorage implements IIdentityStorage {
       await this._profileDataSource.save(element.id, element);
     }
   }
-
+  /**
+   *  @deprecated The method should not be used. It returns only one profile per verifier, which can potentially restrict business use cases
+   *   Use getProfilesByVerifier instead.
+   */
   async getProfileByVerifier(verifier: string): Promise<Profile | undefined> {
     return this._profileDataSource.get(verifier, 'verifier');
   }
 
+  async getProfilesByVerifier(verifier: string, tag?: string): Promise<Profile[]> {
+    return (await this._profileDataSource.load()).filter(
+      (p) => p.verifier === verifier && (!tag || p.tag?.includes(tag))
```

Comment:
Shouldn't we check for strict equality?

`(!tag || p.tag === tag)`

Reply:
no